apps/core-v2-testnet/scripts/watch.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  SnapshotRestorer,
+  takeSnapshot,
+} from "@nomicfoundation/hardhat-network-helpers";
+import { ethers } from "hardhat";
+import prompts from "prompts";
+
+// util function to make a function recursive
+const forever = (func: () => void) =>
+  (async function fn() {
+    await func();
+    fn();
+  })();
+
+async function main() {
+  const provider = ethers.provider;
+
+  // stack to store snapshots
+  let stack: SnapshotRestorer[] = [];
+
+  // register event listener to new blocks
+  provider.on("block", async (block: number) => {
+    const snapshot = await takeSnapshot();
+    stack.push(snapshot);
+  });
+
+  await forever(async () => {
```

Comment:
minor: if we can, let's switch this to just `setInterval(cb, 0)`. Simpler I think, and gives us the ability to stop it w/ `clearInterval` if we wanted.

Reply:
this runs unbounded, since a new stream is getting created every time the interval is called